index.js

```diff
@@ -134,11 +134,40 @@ var Router = function(){
 
       var route = Route(path);
       route.fn = fn;
+      route.idx = this.routes.length;
 
       this.routes.push(route);
       this.routeMap[path] = fn;
     },
 
+    removeRoute: function(path) {
+      if (!path) throw new Error(' route requires a path');
+      if (!this.routeMap[path]) {
+        throw new Error('path does not exist: ' + path);
+      }
+
+      // find route for this specific path, not necessarily the first that
+      // matches the pattern
+      var match;
+      this.routes.forEach(function eachRoute(route) {
+        if (route.src === path) {
+          match = route;
+        }
+      });
+
+      if (match) {
+        this.routes.splice(match.idx, 1);
+        delete this.routeMap[path];
+
+        // re-index
+        for (var i = match.idx; i<this.routes.length; i++) {
```

Comment:
This won't work. `x` is a route object while `v` is a string in your example.

Also, if I don't re-index, future calls to `removeRoute` would behave in unexpected ways. 

I think the biggest misconception here is that I can't call `this.routes.indexOf` without a valid route object and I only have a path. One would think that a call to `match(path)` would return the route to remove, but that does not guarantee to return the route matching the path, but simply the first path in the routes list.